src/libcore/num/mod.rs

```diff
@@ -2523,42 +2524,91 @@ macro_rules! same_sign_try_from_int_impl {
-same_sign_try_from_int_impl!(u128, u32, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i32, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, u64, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i64, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, u128, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i128, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, usize, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, isize, i8, i16, i32, i64, i128, isize);
-
-macro_rules! cross_sign_from_int_impl {
-    ($unsigned:ty, $($signed:ty),*) => {$(
+macro_rules! same_sign_try_from_wider_impl {
+    ($target:ty, $($source:ty),*) => {$(
         #[unstable(feature = "try_from", issue = "33417")]
-        impl TryFrom<$unsigned> for $signed {
+        impl TryFrom<$source> for $target {
             type Error = TryFromIntError;
 
-            fn try_from(u: $unsigned) -> Result<$signed, TryFromIntError> {
-                let max = <$signed as FromStrRadixHelper>::max_value() as u128;
-                if u as u128 > max {
+            #[inline]
+            fn try_from(u: $source) -> Result<$target, TryFromIntError> {
+                let min = <$target>::min_value() as $source;
+                let max = <$target>::max_value() as $source;
+                if u as $source < min || u as $source > max {
                     Err(TryFromIntError(()))
                 } else {
-                    Ok(u as $signed)
+                    Ok(u as $target)
                 }
             }
         }
+    )*}
+}
+
+/// TryFrom on types where the conversion will always succeed.
+macro_rules! trivial_try_from_impl {
+    ($source:ty, $($target:ty),*) => {$(
+        #[unstable(feature = "try_from", issue = "33417")]
+        impl TryFrom<$source> for $target {
+            type Error = TryFromIntError;
+
+            #[inline]
+            fn try_from(u: $source) -> Result<$target, TryFromIntError> {
+                // Using into as an extra check on the macro arguments.
+                Ok(u as $target)
+            }
+        }
+    )*}
+}
 
+// The current implementations assume that usize/isize <= 64 bits wide.
+// (source, $(target))
+trivial_try_from_impl!(u8, u8, u16, i16, u32, i32, u64, i64, u128, i128, usize);
+trivial_try_from_impl!(u16, u16, u32, i32, u64, i64, u128, i128);
+trivial_try_from_impl!(u32, u32, u64, i64, u128, i128);
+trivial_try_from_impl!(u64, u64, u128, i128);
+trivial_try_from_impl!(u128, u128);
+trivial_try_from_impl!(usize, usize, u128, i128);
+
+trivial_try_from_impl!(i8, i8, i16, i32, i64, i128, isize);
+trivial_try_from_impl!(i16, i16, i32, i64, i128);
+trivial_try_from_impl!(i32, i32, i64, i128);
+trivial_try_from_impl!(i64, i64, i128);
+trivial_try_from_impl!(i128, i128);
+trivial_try_from_impl!(isize, isize, i128);
+trivial_try_from_impl!(usize, u64);
```

Comment:
Ah, simply an oversight, you're right it should be together with the other unsigned ones.